metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,2287 @@
+from sqlalchemy import create_engine, inspect
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy.exc import ProgrammingError
+from sqlalchemy.sql import sqltypes as types
+
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter.mce_builder import (
+    make_data_platform_urn,
+    make_dataplatform_instance_urn,
+    make_dataset_urn_with_platform_instance,
+    make_domain_urn,
+    make_tag_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.mcp_builder import (
+    DatabaseKey,
+    PlatformKey,
+    SchemaKey,
+    
+    add_dataset_to_container,
+    add_domain_to_entity_wu,
+    gen_containers,
+    wrap_aspect_as_workunit
+)
+import datahub.emitter.mce_builder as builder
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.ge_profiling_config import GEProfilingConfig
+from datahub.ingestion.source.state.sql_common_state import (
+    BaseSQLAlchemyCheckpointState,
+)
+from datahub.ingestion.source.state.stale_entity_removal_handler import (
+    StaleEntityRemovalHandler,
+    StaleEntityRemovalSourceReport,
+    StatefulStaleMetadataRemovalConfig,
+)
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfigBase,
+    StatefulIngestionSourceBase,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.common import StatusClass
+from datahub.metadata.com.linkedin.pegasus2avro.dataset import UpstreamLineage
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    ArrayTypeClass,
+    BooleanTypeClass,
+    BytesTypeClass,
+    DateTypeClass,
+    EnumTypeClass,
+    ForeignKeyConstraint,
+    MySqlDDL,
+    NullTypeClass,
+    NumberTypeClass,
+    RecordTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringTypeClass,
+    TimeTypeClass,
+)
+from datahub.metadata.schema_classes import (
+    ChangeTypeClass,
+    DataPlatformInstanceClass,
+    DatasetLineageTypeClass,
+    DatasetPropertiesClass,
+    GlobalTagsClass,
+    SubTypesClass,
+    TagAssociationClass,
+    UpstreamClass,
+    ViewPropertiesClass,
+    MLModelDeploymentPropertiesClass,
+    MLModelPropertiesClass
+)
+from datahub.telemetry import telemetry
+from datahub.utilities.lossy_collections import LossyList
+from datahub.utilities.registries.domain_registry import DomainRegistry
+from datahub.utilities.sqlalchemy_query_combiner import SQLAlchemyQueryCombinerReport
+
+if TYPE_CHECKING:
+    from datahub.ingestion.source.ge_data_profiler import (
+        DatahubGEProfiler,
+        GEProfilerRequest,
+    )
+    
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+MISSING_COLUMN_INFO = "missing column information"
+
+
```

Comment:
This entire file seems to have been copy-pasted from other places. Please refactor

Reply:
it was done to avoid code conflict while writing new vertica-specific features.

Reply:
You should avoid code conflict by overriding specific methods or fields, not copying.